VVResonances/interactive/makeCard.py

```diff
@@ -16,85 +16,63 @@
-    card.addMjetBackgroundShapeVJetsRes("Vjets_mjetRes_l2","MJ2","",JJ_VJets__Res_l2,{'CMS_scale_prunedj':1},{'CMS_res_prunedj':1.0})
+    card.addMjetBackgroundShapeVJetsGaus("Vjets_mjetRes_l1","MJ1","",JJ_VJets__Res_l1,{'CMS_scale_prunedj':1},{'CMS_res_prunedj':1.0})
+    card.addMjetBackgroundShapeVJetsGaus("Vjets_mjetRes_l2","MJ2","",JJ_VJets__Res_l2,{'CMS_scale_prunedj':1},{'CMS_res_prunedj':1.0})
     card.product3D("Vjet","Vjets_mjetRes_l1","Vjets_mjetRes_l2","Vjets_mjj")
-    card.addFixedYieldFromFile("Vjet",1,"JJ_VJets_HPHP_jecv6.root","VJets",JJ_VJets__ratio_l1)
-        
-        
-    # add systematics for mjj part of resoant vjets contribution
-    card.addSystematic("CMS_VV_JJ_p0_Vjets_mjj_JJ_WprimeWZ_"+p+"_13TeV","param",[JJ_VJets__MVV['p0']['val'],JJ_VJets__MVV['p0']['err']*20])
-    card.addSystematic("CMS_VV_JJ_p1_Vjets_mjj_JJ_WprimeWZ_"+p+"_13TeV","param",[JJ_VJets__MVV['p1']['val'],JJ_VJets__MVV['p1']['err']*10])
-    card.addSystematic("CMS_VV_JJ_p2_Vjets_mjj_JJ_WprimeWZ_"+p+"_13TeV","param",[JJ_VJets__MVV['p2']['val'],JJ_VJets__MVV['p2']['err']*10])
-    # end V+jets background ####################################################
-
+    card.addFixedYieldFromFile("Vjet",1,"JJ_VJets_%s.root"%p,"VJets",1.0)
 
+    #QCD
+    rootFile="JJ_nonRes_3D_"+p+".root"
+    card.addHistoShapeFromFile("nonRes",["MJ1","MJ2","MJJ"],rootFile,"histo",['PTXY:CMS_VV_JJ_nonRes_PTXY','OPTXY:CMS_VV_JJ_nonRes_OPTXY','PTZ:CMS_VV_JJ_nonRes_PTZ','OPTZ:CMS_VV_JJ_nonRes_OPTZ'],False,0)    
+    card.addFixedYieldFromFile("nonRes",2,"JJ_"+p+".root","nonRes")
 
     #DATA
     card.importBinnedData("JJ_"+p+".root","data",["MJ1","MJ2","MJJ"])
-    
+
     #SYSTEMATICS
 
-     #luminosity
+    #luminosity
     card.addSystematic("CMS_lumi","lnN",{'%s':1.026})
 
-   #kPDF uncertainty for the signal
+    #kPDF uncertainty for the signal
     card.addSystematic("CMS_pdf","lnN",{'%s':1.01})
 
-   #W+jets cross section in acceptance-dominated by pruned mass
+    #background normalization
     card.addSystematic("CMS_VV_JJ_nonRes_norm_"+p,"lnN",{'nonRes':1.5})
-   
     card.addSystematic("CMS_VV_JJ_Vjets_norm_"+p,"lnN",{'Vjet':1.02})
-   
-
-   #tau21 
-    if p=='HPHP':
-       card.addSystematic("CMS_VV_JJ_tau21_eff","lnN",{'%s':1+0.14})
-    if p=='HPLP':
-       card.addSystematic("CMS_VV_JJ_tau21_eff","lnN",{'%s':1-0.33})
-               
+
+    #tau21 
+    if p=='HPHP': card.addSystematic("CMS_VV_JJ_tau21_eff","lnN",{'BulkGWW':1+0.14})
+    else: card.addSystematic("CMS_VV_JJ_tau21_eff","lnN",{'BulkGWW':1-0.33})
+
     #pruned mass scale    
     card.addSystematic("CMS_scale_j","param",[0.0,0.02])
     card.addSystematic("CMS_res_j","param",[0.0,0.05])
     card.addSystematic("CMS_scale_prunedj","param",[0.0,0.02])
     card.addSystematic("CMS_res_prunedj","param",[-0.2,0.001])
 
-    # #alternative shapes
+    #dijet function parameters for V+jets
+    card.addSystematic("CMS_VV_JJ_p0_Vjets_mjj_JJ_"+p+"_13TeV","param",[JJ_VJets__MVV['p0']['val'],1.0)
```

Comment:
in these lines the closing bracket is missing: [JJ_VJets__MVV['p0']['val'],1.0 -> [JJ_VJets__MVV['p0']['val'],1.0]

Reply:
Right! Let me fix this.